### ISSUE TITLE
Uncaught exception when creating message

### ISSUE THREAD

**Issue by jwpoole** [Bandwidth/php-sdk#7]
Migrating to the new SDK from Catapult for SMS message sending. Used 'https://support.bandwidth.com/hc/en-us/articles/360033658613-Messaging-V2-API-Account-Setup-UI-' to find requisite credentials. Attempt to create a message fires an uncaught exception in JsonMapper.

Implementation:
```
$cred = new BandwidthLib\Configuration([
		'messagingBasicAuthUserName' =>  BANDWIDTH_API_TOKEN,
		'messagingBasicAuthPassword' => BANDWIDTH_API_SECRET,
	]
);
try
{
	$messagingClient = $client->getMessaging()->getClient();
	$body = new BandwidthLib\Messaging\Models\MessageRequest();
	$body->from = $from;
	$body->to = ['+1' . $to];
	$body->applicationId = BANDWIDTH_APPLICATION_ID;
	$body->text = $message;

	$response = $messagingClient->createMessage(BANDWIDTH_ACCOUNT_ID, $body);
	print_r($response);
}
catch (Exception $e)
{
	print_r($e);
	return $e;
}
```

Exception:
```
An uncaught Exception was encountered

Type: Error

Message: Class '\BandwidthLib\Models\mixed' not found

Filename: vendor/apimatic/jsonmapper/src/JsonMapper.php

Line Number: 709
```

**Comment by jmulford-bw** [comment]
Hi @jwpoole! Thanks for this. This SDK is autogenerated and we reported this bug to the source.

I was able to recreate this, it looks like this bug occurs on _some_ of the API responses (specifically a malformed phone number, invalid application id, or a 429 rate limit)

**Comment by jwpoole** [comment]
@jmulford-bw any update as to when this will be fixed? We can't tell what issue on our end is causing this and the generic php error is pretty useless for troubleshooting.

**Comment by jmulford-bw** [comment]
Hey @jwpoole we should have an update in the next few days

**Comment by jmulford-bw** [comment]
Hey @jwpoole unfortunately we're expecting a fix for this in about a week. However we may have a workaround to temporarily unblock you. Please open an issue with [Bandwidth Support](support.bandwidth.com) to further discuss this there

**Comment by jwpoole** [comment]
@jmulford-bw I actually already have an issue open with support. The ticket number is 2174378.

**Comment by jmulford-bw** [comment]
Oh, I wasn't aware of that. I apologize. Let's keep this discussion here since that was already established in the ticket.

So I'm gonna make a branch and do a short term fix on that branch, and you can use that. We can do an alpha release on packagist as well if you want to grab that specific version using composer. Whenever our upstream source fixes it on their end we will publish a new release. We can't guarantee that their fix will be identical to this temporary fix, but hopefully this will unblock you in the short term.

If this works for you we can move forward with this

**Comment by jwpoole** [comment]
That sounds great and should work, thanks for all the help.

**Comment by jmulford-bw** [comment]
Cool, here's the release https://github.com/Bandwidth/php-sdk/releases/tag/v1.2.1-alpha and the branch https://github.com/Bandwidth/php-sdk/tree/error-handling-temp-fix. If you're using composer you can grab `v1.2.1-alpha`

I was able to grab the errors with code like this

```
catch (Exception $e)
{
    print_r($e->fieldErrors[0]->description);
    echo "\n";
    print_r($e->fieldErrors[0]->fieldName);
    echo "\n";
    return $e;
}
```

Output: (I used an intentionally malformed application ID to get the response)

```
'cd1f0ef' does not exist or is not owned by user.
applicationId
```

Let me know if this works for you

**Comment by jobyy** [comment]
Any chance the API is documented in OpenAPI or Swagger specification so that one can use AutoRest, Swagger Codegen, OpenAPI Generator, etc to generate the PHP client instead? Thanks.

**Comment by dtolb** [comment]
@jobyy can you open a [support ticket](support.bandwidth.com) and we can communicate there. Thx!

**Comment by jobyy** [comment]
@dtolb done. Request #2283301

**Comment by jmulford-bw** [comment]
Hey @jwpoole upstream has fixed the issue and we just made release 1.2.2 with the fix. I'm gonna close this issue but please re-open it if you still run into it
